contracts/compounder/GenericVault.sol

```diff
@@ -72,11 +73,12 @@ contract GenericUnionVault is ERC20, IERC4626, Ownable, ReentrancyGuard {
     /// @param _reward VirtualBalanceRewardPool address
     /// @return bool success
     function addExtraReward(address _reward) external onlyOwner notToZeroAddress(_reward) returns (bool) {
-        if (extraRewards.length >= 12) {
+        if (extraRewards.length >= 12 || isExtraReward[_reward]) {
```

Comment:
TBF we can probably make this a require. It's `return false` on the BaseRewardPool where this is copied from as to not bork the pool but in this case a require revert makes more sense.

Reply:
Sure thing, just added the revert and revert test 👍